### PR TITLE
Fix error when plotting training history after early stopping.

### PR DESCRIPTION
**Pull request by JamesAllingham**
This PR resolves issue [#117](https://github.com/rstudio/keras/issues/117#issuecomment-326672778). The issue being that plotting of training history would fail when early stopping was used. 

The fix was to pad the training metrics with NAs so that the number of rows is the same as the specified number of training epochs, not the number that occurred before early stopping.

Now when plotting the training history after early stopping we get a plot like this:
early_stopping_plot

There are few things I am a little unsure of:

1. The fix uses NAs which causes the following warning messages to be displayed when calling the `plot()` function:
```
Warning messages:
1: Removed 352 rows containing non-finite values (stat_smooth). 
2: Removed 352 rows containing missing values (geom_point). 
```
Should we hide these or are we happy to have the user see them?

2. Should I add a test case for this issue so that it isn't accidentally broken again in the future?

3. Speaking of tests, I did run the `test_check`, however, both before and after my changes I am getting this output:

```
OK: 209 SKIPPED: 2 FAILED: 0
```
I am assuming that the two skipped tests are due to me only using the TF backend and not CNTK or Theano? Please let me know if this is something I need to have a closer look at.

4. In order to figure out how much padding to add, I used the length of `x$metrics$loss`. My thinking was that there should always be a "loss" in "metrics", so it would be safe to do this - is that a reasonable assumption? 

Let me know what you think and if there are any other changes I need to make.

James

### CODE REVIEW

**Comment by jjallaire** [comment]
Looks good! Just merged.

I got rid of the warnings by adding `na.rm = TRUE` to `geom_smooth` and `geom_point`.

The skipped tests are likely due to running a slightly earlier version of keras rather than related to backends.

**Comment by JamesAllingham** [comment]
Great - I am happy to have contributed (ever so slightly) to this awesome project!

By the way, maybe you could give me some advice for the next time I do this. They way I was testing my changes was that I made a commit, pushed it to my fork for the repo, and then used dev tools to install the keras package from my repo. I then ran the test code. This seemed like a very inefficient and clunky way to do it - is there a better way?

**Comment by jjallaire** [comment]
You should be able to just work within your locally checked out branch and then just call `devtools::load_all()` to iteratively test changed code and `devtools::test()` to run the tests. If you are using RStudio you could also use `Build and Reload` to do something similar to `devtools::load_all`, but with a fresh R session.
